src/plugins/discover/public/application/main/services/discover_app_state_container.ts

```diff
@@ -75,3 +126,155 @@ export interface AppState {
 
 export const { Provider: DiscoverAppStateProvider, useSelector: useAppStateSelector } =
   createStateContainerReactHelpers<ReduxLikeStateContainer<AppState>>();
+
+/**
+ * This is the app state container for Discover main, it's responsible for syncing state with the URL
+ * @param stateStorage
+ * @param savedSearch
+ * @param services
+ */
+export const getDiscoverAppStateContainer = (
```

Comment:
Nit: just a preference, but I typically prefer to pass an options object instead of multiple params in case additional dependencies are needed later. I'm fine with keeping it as it is though if preferred.

Reply:
Agreed, thx!